Compile JNI library Failed

execute:
```make DISABLE_JNI=0```

result:
```
cwei@enderman:~/workspace/dcurl$ make DISABLE_JNI=0
  CC    build/hash/curl.o
  CC    build/constants.o
  CC    build/trinary/trinary.o
  CC    build/dcurl.o
  CC    build/pow_sse.o
src/pow_sse.c: In function ‘pow_sse_init’:
src/pow_sse.c:375:21: warning: implicit declaration of function ‘get_nprocs_conf’ [-Wimplicit-function-declaration]
     DCURL_NUM_CPU = get_nprocs_conf() - 1;
                     ^
make: *** No rule to make target 'build/jni/iri-pearldiver-exlib.o', needed by 'build/test_trinary'.  Stop.
```

--- a/Makefile
+++ b/Makefile
@@ -19,8 +19,9 @@ CFLAGS += \
 LDFLAGS += -L$(OPENCL_LIB) -lOpenCL
 endif
 
+# FIXME: avoid hardcoded OS path
 ifneq ("$(DISABLE_JNI)","1")
-CFLAGS += \
+CFLAGS_JNI = \
 	-I$(OPENJDK_PATH)/include \
 	-I$(OPENJDK_PATH)/include/linux
 endif
@@ -75,6 +76,9 @@ $(OUT)/test_%.o: test/test_%.c
 	$(Q)$(CC) -o $@ $(CFLAGS) -c -MMD -MF $@.d $<
 
 $(OUT)/jni/%.o: jni/%.c
+	$(VECHO) "  CC\t$@\n"
+	$(Q)$(CC) -o $@ $(CFLAGS) $(CFLAGS_JNI) -c -MMD -MF $@.d $<
+
 $(OUT)/trinary/%.o: $(SRC)/trinary/%.c
 $(OUT)/hash/%.o: $(SRC)/hash/%.c
 $(OUT)/%.o: $(SRC)/%.c

